### PR TITLE
SC-8323 - clear collection when querying

### PR DESCRIPTION
**Pull request by Ojself**
# deprecated:
### Duplicated pr, see https://github.com/hpi-schul-cloud/nuxt-client/pull/1632

🚫 Before: When doing a search in the Learning-Store, the previous result was visible for a split second for the user
✅ Now: The previous result is cleared from the state and never shown to the user when using the Learning-Store.

See ticket for visual aid

## Links to Tickets or other pull requests

https://ticketsystem.hpi-schul-cloud.org/browse/SC-8323

## Changes
Collection in Lern-Store will now not render the previous results when doing a new search
The changes are required because it improves user experience

## New Repos, NPM pakages or vendor scripts

One new package "object-hash":
https://www.npmjs.com/package/object-hash
Object hash to store the queries in the state

## Screenshots of UI changes
There's a .mp4 in the ticket showing the result of the new code

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

**Comment by Ojself** [comment]
@schul-cloud-bot update screenshots

**Comment by Ojself** [comment]
@schul-cloud-bot update screenshots

**Review by virgilchiriac — APPROVED** [review]

**Comment by Ojself** [comment]
@schul-cloud-bot update screenshots

**Comment by Ojself** [comment]
@schul-cloud-bot update screenshots

**Comment by Ojself** [comment]
@schul-cloud-bot update screenshots

**Comment by Ojself** [comment]
@schul-cloud-bot update screenshots

**Comment by Ojself** [comment]
@schul-cloud-bot update screenshots

**Comment by virgilchiriac** [comment]
storybook seems to be broken due to the upgrade
http://stories.1626.nuxt.schul-cloud.surge.sh/

File: package.json [hunk 1]
```diff
@@ -99,7 +100,7 @@
 		"@storybook/addon-storysource": "^6.0.28",
 		"@storybook/addon-viewport": "^6.0.28",
 		"@storybook/addons": "^6.0.28",
-		"@storybook/vue": "^5.3.19",
+		"@storybook/vue": "^6.1.11",
```
Comment (virgilchiriac) [inline package.json:103]: looks like the upgrade breaks storybook

**Review by virgilchiriac — CHANGES_REQUESTED** [review]

**Comment by Ojself** [comment]
@schul-cloud-bot update screenshots

**Review by virgilchiriac — APPROVED** [review]

**Comment by Ojself** [comment]
Duplicated: 
https://github.com/hpi-schul-cloud/nuxt-client/pull/1632

**Pull request closed without merging**